File: backend/DTO/CourseDTO.cs
using System.Text.Json.Serialization;

namespace backend.DTO;

public record CourseDTO
{
    [JsonPropertyName("id")]
    public string? ID { get; init; }
    [JsonPropertyName("faculty_code")]
    public required string FacultyCode { get; init; }
    [JsonPropertyName("code")]
    public required string Code { get; init; }
    [JsonPropertyName("icon")]
    public required string Icon { get; init; }
    [JsonPropertyName("name")]
    public required string Name { get; init; }
    [JsonPropertyName("count")]
    public int Count { get; init; } = 0;

    [JsonPropertyName("created_at")]
    public DateTime? CreatedAt { get; init; }
}

File: frontend/src/api/dto/course.dto.ts
import { Course } from "@/src/types";

export type CourseDTO = {
  id: string;
  faculty_code: string;
  code: string;
  icon: string;
  name: string;
  created_at: string;
};

export const parseCourseDTO = (data: CourseDTO): Course => ({
  id: data.id,
  facultyCode: data.faculty_code,
  code: data.code,
  icon: data.icon,
  name: data.name,
  createdAt: data.created_at,
});

export const parseCourseDTOList = (data: CourseDTO[]): Course[] => {
  return data.map(parseCourseDTO);
};
